### ISSUE TITLE
Cannot create symbolic links on drvfs network share

### ISSUE THREAD

**Issue by mfroment** [microsoft/WSL#5524]
# Environment
```none
Windows build number: Microsoft Windows [Version 10.0.19041.329] 
Your Distribution version: Ubuntu Release:        20.04
Issue in WSL2
```

# Steps to reproduce
I have 2 PCs with same WSL2+ubuntu set up, and sharing my home directory as \\\\PC{1,2}\WSL_home
Mounting the remote home dir in WSL works fine (permissions, ownership, file & directory CRUD, symlink following), except for symbolic link creation.
```
$ sudo mount -t drvfs '\\PC2\WSL_home\'  /mnt/rhome -o metadata,uid=1001,gid=1001
$ cd /mnt/rhome
$ ln -s /mnt/c/Users/MyUserName nthome
ln: failed to create symbolic link 'nthome': Operation not permitted
$ touch dummyfile
$ ln -s dummyfile dummylink
ln: failed to create symbolic link 'dummylink': Operation not permitted
```
I have tried to mount local pc share (\\\\localhost\\WSL_home\) and do same steps as above, same outcome.

I have also tried combinations of running WSL as administrator or not, granting (or not) Symbolic Link Creation by policy, enabling (or not) symbolic link evaluations for remote to local & remote to remote symbolic link ; this on local and/or remote hosts. Same outcome.

[Strace gist for the dummyfile -> dummylink example, with \\\\localhost\\WSL_home\ mounted as /mnt/rhome](https://gist.github.com/mfroment/d41736ca59254ebf0ba9ebc577501ebf)

#  Expected behavior

(WSL) symbolic link to be created.

# Actual behavior

Cannot create symbolic link on mounted network share.

**Comment by therealkenc** [comment]
This is same #2144 (answer: [message](https://github.com/Microsoft/WSL/issues/2144#issuecomment-307965136)). I suspect but can't prove that answer stands, considering #4868

> Issue in WSL2

On WSL2 you can do:

```
$ sudo mount -t cifs -o username=julius,password=password,uid=1000,gid=1000,vers=3.0,mfsymlinks //192.168.48.1/Users/julius ./mnt
```

![image](https://user-images.githubusercontent.com/7049700/86381226-ec34c980-bc41-11ea-9505-ca12866fce2e.png)

**Comment by mfroment** [comment]
Thanks for the heads up on mounting with cifs.
That does allow creating symlinks on the shared drive.

Problem is that those are Minshall+French links, not WSL symbolic links (or any kind of reparse point).
They do not work on a WSL session run on the 2nd PC (they appear as plain files "XSym ... ").

**Comment by therealkenc** [comment]
> But those are Minshall+French links

Yeah, it isn't worth much and doesn't satisfy your ask. I was just throwing it out there.

**Comment by scruel** [comment]
$ sudo mount -t cifs -o uid=1000,gid=1000,vers=3.0,mfsymlinks N: /mnt/n
mount: /mnt/n: bad option; for several filesystems (e.g. nfs, cifs) you might need a /sbin/mount.<type> helper program.
